perma_web/perma/celery_tasks.py

```diff
@@ -1366,21 +1391,29 @@ def format_filesize(i):
             "raw_warc_save_duration": warc_save_duration,
             "raw_pages_write_duration": pages_write_duration,
             "raw_conversion_duration": conversion_duration,
-            "error": ''
+            "error": '',
+            "warc_checksums_match": warc_checksums_match
         }
-        writer = csv.DictWriter(log_file, fieldnames=row.keys())
-
         if conversion_error:
             row["conversion_status"] = "Failure"
             row["error"] = error_output
-            writer.writerow(row)
-        elif wacz_size == 0 or warc_size > wacz_size:
+        elif wacz_size == 0:
             row["conversion_status"] = "Failure"
-            row["error"] = "WACZ is smaller than WARC"
-            writer.writerow(row)
+            row["error"] = "No WACZ produced"
         else:
-            row["conversion_status"] = "Success"
-            writer.writerow(row)
+            if warc_size > wacz_size:
+                row["conversion_status"] = "Failure"
+                row["error"] = "WACZ is smaller than WARC"
+            elif not warc_checksums_match:
+                row["conversion_status"] = "Failure"
+                row["error"] = "The WARC embedded in the WACZ differs from the source WARC"
+            else:
+                row["conversion_status"] = "Success"
```

Comment:
Yes: if we are in the 1403 `else`, it means there is a WACZ to save to storage (either because it came out well, or because we want to study how it didn't). In the other cases, no WACZ is present.

I can refactor or comment to make this clearer.